### PR TITLE
Implement Azure e2e template tests

### LINKED ISSUE #210: E2e verification of Azure templates
Should be based on #201

### PR DESCRIPTION
**Pull request by kylewuolle**
Implement Azure template e2e tests. Closes #210.

### CODE REVIEW

**Comment by squizzi** [comment]
You'll need to reopen this with a repository branch and modify the test to run as `pull_request` or just rebase off my PR, #280 since there's a bunch of test related fixes in there.  Then you'll be able to see if the tests pass and iterate on them before we merge it in as `pull_request_target`.  This is the best way I can think of to do this until we have more time to figure out the secrets stuff in actions.

File: test/kubeclient/kubeclient.go
Comment (squizzi) [inline test/kubeclient/kubeclient.go:194]: You don't even need to call `.Get` prior to `.Create` you can just call `.Create` and then check for `apierrors.IsAlreadyExists(err)` and return `nil` in that case.

File: test/kubeclient/kubeclient.go
Comment (squizzi) [inline test/kubeclient/kubeclient.go:158]:
```suggestion
			if errors.Is(err, io.EOF) {
```

File: test/kubeclient/kubeclient.go
Comment (squizzi) [inline test/kubeclient/kubeclient.go:142]: If you're going to perform substitutions on the file it should probably be a `.tpl` file.  Perhaps create an `resources/azure-credentials.yaml.tpl` and embed it above and then call `envsubst` on it.

File: test/kubeclient/kubeclient.go
Comment (squizzi) [inline test/kubeclient/kubeclient.go:136]: Instead of calling `os.ReadFile` on a path that might change use `go:embed` to embed the file at runtime, see: https://gobyexample.com/embed-directive you can embed directly into `[]byte` and then manipulate it as needed below.

File: test/managedcluster/managedcluster.go
Comment (squizzi) [inline test/managedcluster/managedcluster.go:70]: Note that this name doesn't align with the one ran in CI, if you want to change that name to make sure you modify `MANAGED_CLUSTER_NAME` in the actions workflow when you rebase.

File: test/managedcluster/validate_deleted.go
Comment (squizzi) [inline test/managedcluster/validate_deleted.go:46]: 👍

**Review by squizzi — CHANGES_REQUESTED** [review]

File: test/e2e/e2e_test.go
Comment (a13x5) [inline test/e2e/e2e_test.go:203]: You're creating hosted cluster template in the local kind cluster. It should fail. And if it's not - the test should be reworked

File: test/kubeclient/kubeclient.go
Comment (a13x5) [inline test/kubeclient/kubeclient.go:134]: It looks more complex than it should be. @squizzi already proposed several fixes and I tend to agree with him

**Review by a13x5 — CHANGES_REQUESTED** [review]
The hosted test looks incorrect, should be rechecked and reworked. 
Since hosted cluster should be create on cluster which is deployed on Azure. Also artifacts should be passed there somehow. I don't see any of that. 
Probably you should wait for #280 to be merged and then build hosted CP tests on top of that.

**Pull request closed without merging**
